mods/toggle-taskbar-autohide.wh.cpp

```diff
@@ -0,0 +1,1108 @@
+    if (pUIAutomation == NULL) {
+        Wh_Log(L"UIAutomationWrapper failed to get instance\n");
+        return false;
+    }
+
+    IUIAutomationElement *pWindowElement = NULL;
+    HRESULT hr = pUIAutomation->ElementFromPoint(pointerLocation, &pWindowElement);
+    if (FAILED(hr) || (pWindowElement == NULL)) {
+        Wh_Log(L"Failed to retrieve UI element from mouse click\n");
+        return false;
+    }
+
+    BSTR className = NULL;
+    hr = pWindowElement->get_CurrentClassName(&className);
+    if (FAILED(hr) || (className == NULL)) { 
+        Wh_Log(L"Failed to retrieve the Name of the UI element clicked.");
+        pWindowElement->Release();
+        return false;
+    }
+    Wh_Log(L"Clicked UI element ClassName: %s", className);
+    const bool taskbarClicked = (wcscmp(className, L"Shell_TrayWnd") == 0) || // Windows 10 primary taskbar
+                                (wcscmp(className, L"Shell_SecondaryTrayWnd") == 0) ||   // Windows 10 secondary taskbar
+                                (wcscmp(className, L"Taskbar.TaskbarFrameAutomationPeer") == 0) ||     // Windows 11 taskbar
+                                (wcscmp(className, L"Windows.UI.Input.InputSite.WindowClass") == 0);   // Windows 11 21H2 taskbar
+    SysFreeString(className);
+    pWindowElement->Release();
+
+    if (taskbarClicked) {
+        const bool isAutohideEnabled = ToggleTaskbarAutohide();
+        Wh_Log(L"Setting taskbar autohide state to %s", isAutohideEnabled ? L"enabled" : L"disabled");
+    }
+
+    return false;
+}
+
+////////////////////////////////////////////////////////////
+
+BOOL Wh_ModInit() {
+    Wh_Log(L">");
+
+    LoadSettings();
+
+    if (!WindowsVersionInit() || (g_taskbarVersion == UNKNOWN_TASKBAR)) {
+        Wh_Log(L"Unsupported Windows version");
+        return FALSE;
+    }
+
+    // treat Windows 11 taskbar as on older windows
+    if ((g_taskbarVersion == WIN_11_TASKBAR) && g_settings.oldTaskbarOnWin11) {
+        g_taskbarVersion = WIN_10_TASKBAR;
+    }
+    Wh_Log(L"Using taskbar version: %s", TaskBarVersionStr[g_taskbarVersion]);
+
+    Wh_SetFunctionHook((void *)CreateWindowExW, (void *)CreateWindowExW_Hook, (void **)&CreateWindowExW_Original);
+
+    HMODULE user32Module = LoadLibrary(L"user32.dll");
+    if (user32Module) {
+        void *pCreateWindowInBand = (void *)GetProcAddress(user32Module, "CreateWindowInBand");
+        if (pCreateWindowInBand) {
+            Wh_SetFunctionHook(pCreateWindowInBand, (void *)CreateWindowInBand_Hook, (void **)&CreateWindowInBand_Original);
+        }
+    }
+
+    WNDCLASS wndclass;
+    if (GetClassInfo(GetModuleHandle(NULL), L"Shell_TrayWnd", &wndclass)) {
+        HWND hWnd = FindCurrentProcessTaskbarWindows(&g_secondaryTaskbarWindows);
+        if (hWnd) {
+            HandleIdentifiedTaskbarWindow(hWnd);
+        }
+    }
+
+    if (!g_UIAutomation.init()) {
+        Wh_Log(L"UIAutomationWrapper failed to initialize\n");
```

Comment:
It should be before the subclass code, otherwise, the subclass won't be unset and Explorer will crash.